js/src/ImageModel.js

```diff
@@ -0,0 +1,86 @@
+/* Copyright 2015 Bloomberg Finance L.P.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var widgets = require("@jupyter-widgets/base");
+var d3 = require("d3");
+var _ = require("underscore");
+var markmodel = require("./MarkModel");
+
+var ImageModel = markmodel.MarkModel.extend({
+
+    defaults: function() {
+        return _.extend(markmodel.MarkModel.prototype.defaults(), {
+            _model_name: "ImageModel",
+            _view_name: "Image",
+            x0: 0.0,
+            y0: 0.0,
+            x1: 1.0,
+            y1: 1.0,
+            scales_metadata: {
+                'x': {'orientation': 'horizontal', 'dimension': 'x'},
+                'y': {'orientation': 'vertical', 'dimension': 'y'},
+            },
+        });
+    },
+
+    initialize: function() {
+        ImageModel.__super__.initialize.apply(this, arguments);
+        this.on("change:x0 change:y0 change:x1 change:y1", this.update_data, this);
```

Comment:
Any reason for not using `on_some_change`here?

Reply:
No, didn't know it existed, let me change that